packages/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/layoutReanimation/TabNavigatorObserver.java

```diff
@@ -0,0 +1,122 @@
+package com.swmansion.reanimated.layoutReanimation;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.FragmentManager;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class TabNavigatorObserver {
+  private final Set<Integer> mFragmentsWithListenerRegistry = new HashSet<>();
+  private final ReaLayoutAnimator mReaLayoutAnimator;
+
+  public TabNavigatorObserver(ReaLayoutAnimator reaLayoutAnimator) {
+    mReaLayoutAnimator = reaLayoutAnimator;
+  }
+
+  public void handleScreenContainerUpdate(View screen) {
+    try {
+      Class<?> screenClass = screen.getClass();
+      Method getScreenFragment = screenClass.getMethod("getFragment");
+      Fragment fragment = (Fragment) getScreenFragment.invoke(screen);
+      int fragmentTag = fragment.getId();
+      if (!mFragmentsWithListenerRegistry.contains(fragmentTag)) {
+        mFragmentsWithListenerRegistry.add(fragmentTag);
+        fragment
+            .getParentFragmentManager()
+            .registerFragmentLifecycleCallbacks(new FragmentLifecycleCallbacks(fragment), true);
+      }
+    } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+      String message = e.getMessage() != null ? e.getMessage() : "Unable to get screen fragment";
+      Log.e("[Reanimated]", message);
+    }
+  }
+
+  class FragmentLifecycleCallbacks extends FragmentManager.FragmentLifecycleCallbacks {
+    private View firstScreen;
+    private final Method getScreen;
+    private final Method getActivityState;
+    private final Set<Integer> screenTagsWithListener = new HashSet<>();
+    private final List<View> nextTransition = new ArrayList<>();
+
+    public FragmentLifecycleCallbacks(Fragment fragment)
+        throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
```

Comment:
Probably we could pass the values the constructor needs so it wouldn't fail, but that's more of an aesthetic decision in this case.

Reply:
I understand your concerns, but the constructor must ensure that a constructed object is valid. In this case, it will not, so I would need to add additional checks in every method within this class to verify if the object was properly initialized.